feat(builder-bob): dont use npm bin to get tsc path

atlj opened the pull request:
## Summary

Closes #434
The builder bob typescript build phase uses `yarn bin` and `npm bin` commands to locate the `tsc`. However, the `npm bin` command is removed with npm 9.0. Please see: https://github.blog/changelog/2022-10-24-npm-v9-0-0-released/

## Test plan

> Both of the test cases start with the following steps:
1. Clone the repo and cd into `packages/react-native-builder-bob`
2. Run `yarn prepack` to build the builder bob.
3. Run `npm link`
4. Create a new project with `npx create-react-native-library@latest my-awesome-project`. You can select `javascript` as the type.
5. Run `npm link react-native-builder-bob` to link the project.

### Resolve `tsc` automatically

1. Run `npm run prepack` to verify the typescript compilation step works properly.
2. Run `yarn prepack` to verify the typescript compilation step works properly.
3. Run `pnpm prepack` to verify the typescript compilation step works properly.

### Resolve `tsc` using the builder bob config

1. Go to the `package.json` file of the project you've generated.
2. Go to the `react-native-builder-bob` section and specify the tsc binary as following: (the `tsc` line needs to be added.)
```json
  "react-native-builder-bob": {
    "source": "src",
    "output": "lib",
    "targets": [
      "commonjs",
      "module",
      [
        "typescript",
        {
          "project": "tsconfig.build.json",
          "tsc": "./node_modules/.bin/tsc"
        }
      ]
    ]
  }
```
3. Run `npm run prepack` to verify the typescript compilation step works properly.
4. Run `yarn prepack` to verify the typescript compilation step works properly.
5. Run `pnpm prepack` to verify the typescript compilation step works properly.

satya164 reviewed (changes requested):
I think it's important to ensure that we're running the local version of `tsc`. Using a global version of `tsc` isn't desired and may produce in inconsistent output. That's why there was a warning previously.

Using `npx` like this is problematic as it will try to install the package if it's not installed (the `tsc` package isn't even the official one), or run the global version if it's not present locally. I'm also not sure how it'll work in yarn monorepo (or regular projects using yarn).

The test plan also mentions that `tsc` option needs to be specified in bob's config - so I'm not sure if `npx` is working here as specifying the config skips the logic that uses `npx`.

This change needs to keep the previous behavior:

- Make sure that it executes locally installed `tsc` bindary
- If it's not installed locally, either use global version with a warning, or throw an error
- Make sure that this change works in repos using yarn, pnpm, etc.,

atlj commented:
I have removed the `npx` logic and now just using a bare path to determine where tsc is.
This is due to two reasons:
1. `npm bin` is deprecated and [it is not aware of workspaces](https://docs.npmjs.com/cli/v8/commands/npm-bin?v=true#synopsis)
2. npm binaries are always linked under `node_modules/.bin`. Please see [this](https://docs.npmjs.com/cli/v9/configuring-npm/folders?v=true#executables).

>The test plan also mentions that tsc option needs to be specified in bob's config - so I'm not sure if npx is working here as specifying the config skips the logic that uses npx.

> Make sure that this change works in repos using yarn, pnpm, etc.,

I've updated the testing section to support multiple cases
